src/core/Meganav/MeganavPanel.tsx

```diff
@@ -0,0 +1,123 @@
+import React from "react";
+import cn from "../utils/cn";
+import Icon from "../Icon";
+import { FlyoutPanelHighlight, FlyoutPanelList } from "./data";
+import FeaturedLink from "../FeaturedLink";
+import { ProductName, products } from "../ProductTile/data";
+import ProductTile from "../ProductTile";
+
+export const MeganavPanel = ({
+  displayProductTile,
+  panelLeft,
+  panelLeftClassName,
+  panelRightHeading,
+  panelRightItems,
+  panelRightBottom,
+}: {
+  displayProductTile?: boolean;
+  panelLeft?: FlyoutPanelHighlight;
+  panelLeftClassName?: string;
+  panelRightHeading?: string;
+  panelRightItems: FlyoutPanelList[];
+  panelRightBottom?: React.ReactNode;
+}) => {
+  return (
+    <>
+      <div className="flex flex-col md:flex-row gap-x-24 bg-neutral-000 dark:bg-neutral-1300">
+        <div
+          className={cn(
+            "flex-[7] flex-shrink-0 grid-cols-1 xs:grid-cols-2 group",
+            panelLeftClassName,
+          )}
+          onClick={
+            !displayProductTile
+              ? () => (window.location.href = panelLeft?.url as string)
+              : undefined
+          }
+        >
+          {displayProductTile
+            ? [
+                "pubsub",
+                "spaces",
+                "chat",
+                "liveSync",
+                "assetTracking",
+                "liveObjects",
+              ].map((product) => (
+                <ProductTile
+                  name={product as ProductName}
+                  key={product}
+                  selected={false}
+                  onClick={() =>
+                    (window.location.href = products[product as ProductName]
+                      ?.link as string)
```

Comment:
Same as my other point earlier, it's best to not make a type assertion on something that we don't know will always be as we expect.